public/js/components/EditorBreakpoint.js

```diff
@@ -26,14 +26,25 @@ const Breakpoint = React.createClass({
 
   displayName: "Breakpoint",
 
-  componentWillMount: function() {
+  addBreakpoint: function() {
     const bp = this.props.breakpoint;
     const line = bp.location.line - 1;
-
     this.props.editor.setGutterMarker(line, "breakpoints", makeMarker());
     this.props.editor.addLineClass(line, "line", "breakpoint");
```

Comment:
This function will now fire twice whenever the editor is rerendered (and thus the breakpoints). This could be solved by implementing `shouldComponentUpdate`:

```js
shouldComponentUpdate(nextProps) {
  return this.props.editor !== nextProps.editor;
}
```

Now, it will only add the breakpoint if the editor changes (which will happen if it was `null` before).